addons/mail/static/src/core/common/composer.js

```diff
@@ -569,9 +569,13 @@ export class Composer extends Component {
     async processMessage(cb) {
         const el = this.ref.el;
         const attachments = this.props.composer.attachments;
-        if (
+        if (attachments.some(({ uploading }) => Boolean(uploading))) {
+            this.env.services.notification.add(_t("Please wait while the file is uploading."), {
+                type: "warning",
+            });
```

Comment:
Just a thought about notifications, not to be done in this fix: we never close previous notifications. I think it's bloating the interface fast. We don't use Discuss much, but I bet live chat users always have their screens bloated with notifications. I think we should rethink the way we display them. Maybe limit our footprint to one notification of each type, replacing the existing ones?


![image](https://github.com/user-attachments/assets/eed4cb02-31fe-4ad6-bf54-e99e8262c33a)
![image](https://github.com/user-attachments/assets/b9757b6c-23be-42cd-83c8-2be0cadc7e6b)

Reply:
Agree, the notification service is quite rudimentary as it is, and it shows this is intended to be used with a low frequency. Discuss can challenge its rudimentary design as it is.

I think there should be a notification hub. The messaging menu acts a bit like one but not really, and notification service is kinda that but not really too...